dask/cli.py

```diff
@@ -0,0 +1,82 @@
+from importlib.metadata import entry_points
+
+from dask import __version__
+
+try:
+    import click
+except ImportError as e:
+    msg = (
+        "The dask.cli module requires click to be installed.\n\n"
+        "Install with conda or pip:\n\n"
+        " conda install click\n"
+        " pip install click\n"
+    )
+    raise ImportError(msg) from e
+
+
+CONTEXT_SETTINGS = {
+    "help_option_names": ["-h", "--help"],
+    "max_content_width": 88,
+}
+
+
+@click.group(context_settings=CONTEXT_SETTINGS)
+@click.version_option(__version__)
+def cli():
+    """Dask command line interface."""
+    pass
+
+
+@cli.command()
+def docs():
+    """Open Dask documentation (https://docs.dask.org/) in a web browser."""
+    import webbrowser
+
+    webbrowser.open("https://docs.dask.org")
+
+
+@cli.group()
+def info():
+    """Information about your dask installation."""
+    pass
+
+
+@info.command()
+def versions():
+    """Print versions of Dask related projects."""
+    from dask.utils import show_versions
+
+    show_versions()
+
+
+def _register_command_ep(interface, entry_point):
+    """Add `entry_point` command to `interface`.
+
+    Parameters
+    ----------
+    interface : click.Command or click.Group
+        The click interface to augment with `entry_point`.
+    entry_point : importlib.metadata.EntryPoint
+        The entry point which loads to a ``click.Command`` or
+        ``click.Group`` instance to be added as a sub-command or
+        sub-group in `interface`.
+
+    """
+    command = entry_point.load()
+    if not isinstance(command, (click.Command, click.Group)):
+        raise TypeError(
+            "entry points in 'dask_cli' must be instances of "
+            "click.Command or click.Group"
+        )
+    interface.add_command(command)
+
+
+def run_cli():
+    """Run the dask command line interface."""
+
+    # discover "dask_cli" entry points and try to register them to the
+    # top level `cli`.
+    for ep in entry_points(group="dask_cli"):
```

Comment:
It looks like the `group=` kwarg is new in Python `3.10`. `importlib.metadata` changed a bunch in `3.10` but is now stable.

https://docs.python.org/3/library/importlib.metadata.html

`3.8`/`3.9` don't take any arguments and returns a `dict`. In `3.10` it returns an `Entrypoints` object which has a `select()` method that takes a `group` kwarg and all kwargs passed to `entry_points(**kwargs)` are passed on to `select()`.



```python
# python 3.8/3.9
importlib.metadata.entry_points().get("dask_cli", [])

# python 3.10
importlib.metadata.entry_points(group="dask_cli")
# also the equivalent of
importlib.metadata.entry_points().select(group="dask_cli")
```

Given that we support `3.8+` we probably need some compatibility code here.

Reply:
Maybe something like

```python
try:
    _eps = entry_points(group="dask_cli")
except TypeError:
    _eps = entry_points().get("dask_cli", [])
for ep in _eps:
    ...
```

Reply:
Now using `from dask.compatibility import entry_points` :+1: